zebra-network/src/peer/connection.rs

```diff
@@ -407,6 +483,8 @@ where
     // context (namely, the work of processing the inbound msg as a request)
     #[instrument(skip(self))]
     async fn handle_message_as_request(&mut self, msg: Message) {
+        // XXX(hdevalence) -- using multiple match statements here
+        // prevents us from having exhaustiveness checking.
```

Comment:
do we want to try to find a way to get exhaustive matching here?

Reply:
Yeah, the way we'd do that is by combining all the match statements into one, but I didn't want to do it in this change because it's already quite large.

Reply:
okay, wanna make an issue for this assuming there isn't one already?

Reply:
Yep, I'll do that after this is merged.